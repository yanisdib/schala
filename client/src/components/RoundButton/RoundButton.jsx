import styled from 'styled-components';


const RoundButton = styled.button`
    width: ${props => props.size ? props.size : 35}px;
    height: ${props => props.size ? props.size : 35}px;
    background-color: ${props => props.backgroundColor ? props.backgroundColor : 'transparent'};
    border: solid 1px ${props => props.borderColor ? props.borderColor : '#000000'};
    border-radius: ${props => props.size ? props.size : 20}px;
    margin: 10px 4px !important;
    cursor: pointer;

    svg{
        height: ${props => props.size ? `calc(${props.size}px - ${props.size / 2})` : '20px'};
        stroke-width: 2px;
    }
`;


export default RoundButton;