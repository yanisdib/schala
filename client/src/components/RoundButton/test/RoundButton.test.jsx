import { shallow } from 'enzyme';

import RoundButton from '../RoundButton';


describe('RoundButton component', () => {
    let wrapper;

    beforeEach(() => wrapper = shallow(<RoundButton>Hello</RoundButton>))

    it('renders correctly without crashing', () => expect(wrapper).toMatchSnapshot())

    it('should receive children', () => expect(wrapper.text()).toEqual('Hello'))
})