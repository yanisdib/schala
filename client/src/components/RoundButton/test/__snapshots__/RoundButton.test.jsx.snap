// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoundButton component renders correctly without crashing 1`] = `
<button
  className="sc-bdfBQB cnLAJQ"
>
  Hello
</button>
`;
